python/cuml/_thirdparty/sklearn/preprocessing/_discretization.py

```diff
@@ -22,29 +22,24 @@
 from cuml.cluster import KMeans
 
 from ..utils.skl_dependencies import BaseEstimator, TransformerMixin
+from cuml.common.mixins import SparseInputTagMixin
 from ..utils.validation import check_is_fitted
 from ..utils.validation import FLOAT_DTYPES
 from ..utils.validation import _deprecate_positional_args
-from ....thirdparty_adapters import check_array, get_input_type, \
-                                    to_output_type
+from ....thirdparty_adapters import check_array
 from ....common.import_utils import check_cupy8
+from ....common.array_sparse import SparseCumlArray
+from ....common.array_descriptor import CumlArrayDescriptor
+from ....common.memory_utils import using_output_type
 
 
 def digitize(x, bins):
-    # With right = Flase and bins in increasing order
-    out = np.full(shape=x.shape, fill_value=0, dtype=np.int32)
-    for i in range(1, len(bins)):
-        bool_arr = np.logical_and(bins[i-1] <= x, x < bins[i])
-        matched = np.where(bool_arr)
-        out[matched] = i
+    return np.searchsorted(bins, x, side='left')
```

Comment:
Woah this wipes out a lot of old code ;) Does this change come from a recent addition of searchsorted to cupy? Just curious why we couldn't use it before.

Reply:
Yes, indeed ;). I don't remember exactly why I implemented this. I think it may have to be due to CuPy only having it implemented in CuPy 8. Anyway, now that all RAPIDS envs should come with CuPy 8.0+, it isn't needed anymore.